pyvrp/cpp/SubPopulation.h

```diff
@@ -0,0 +1,103 @@
+#ifndef SUBPOPULATION_H
+#define SUBPOPULATION_H
+
+#include "Individual.h"
+#include "ProblemData.h"
+#include "diversity/diversity.h"
+
+#include <functional>
+#include <iosfwd>
+#include <stdexcept>
+#include <vector>
+
+struct PopulationParams
+{
+    size_t minPopSize;
+    size_t generationSize;
+    size_t nbElite;
+    size_t nbClose;
+    double lbDiversity;
+    double ubDiversity;
+
+    PopulationParams(size_t minPopSize = 25,
+                     size_t generationSize = 40,
+                     size_t nbElite = 4,
+                     size_t nbClose = 5,
+                     double lbDiversity = 0.1,
+                     double ubDiversity = 0.5)
+        : minPopSize(minPopSize),
+          generationSize(generationSize),
+          nbElite(nbElite),
+          nbClose(nbClose),
+          lbDiversity(lbDiversity),
+          ubDiversity(ubDiversity)
+    {
+        if (0 > lbDiversity || 1 < lbDiversity)
```

Comment:
Are 0 and 1 not allowed? If not the exception should mention the open interval (0, 1) but I think we can do `>=` and `<=` here as well as below?

Reply:
I get that 0 may get you two equal parents but I'm not sure if we need to forbid, even though offspring is the same potentially the local search is randomized (e.g. intensification) so this can still be useful. And I think at least 1 should be allowed)

Reply:
But we do not forbid either zero or one?

Reply:
Although `lbDiversity = 1` will not pass because a condition below this (`lbDiversity <= ubDiversity`) fails.

Reply:
I've flipped the inequalities to hopefully be clearer now.